umbral-pre/src/capsule.rs

```diff
@@ -168,7 +174,7 @@ impl Capsule {
         }
 
         let pub_key = receiving_sk.public_key().to_point();
-        let dh_point = &precursor * &receiving_sk.to_secret_scalar();
+        let dh_point = &precursor * receiving_sk.to_secret_scalar().as_secret();
```

Comment:
Just curious: why not using `&` here on `receiving_sk`? Is it because the new `SecretBox`wrapper?

Reply:
Initially, `&` was applied to the whole `receiving_sk.to_secret_scalar()`, which returns a scalar itself; the arithmetic operator needs a reference. `as_secret()` gives a reference already (which means that the object itself is kept in the heap, wherever the wrapping `Box` put it), and the secret is not copied to the stack as it was before.

Reply:
At least, not copied to the stack in our code. The `*` implementation may still make copies, and that's something we can't prevent.